api/tests/Api/SnapshotTests/ResponseSnapshotTest.php

```diff
@@ -34,20 +39,28 @@ public function testRootEndpointMatchesSnapshot() {
     }
 
     public function testOpenApiSpecMatchesSnapshot() {
-        $response = static::createClientWithCredentials()
-            ->request(
-                'GET',
-                '/docs.json',
-                [
-                    'headers' => [
-                        'accept' => 'application/json',
-                    ],
-                ]
-            )
-        ;
+        $container = static::$kernel->getContainer();
+
+        /** @var OpenApiFactoryInterface $openApiFactory */
+        $openApiFactory = $container->get('api_platform.openapi.factory');
+
+        /** @var NormalizerInterface $normalizerInterface */
+        $normalizerInterface = $container->get('api_platform.openapi.normalizer.api_gateway');
+
+        $openApiResult = $openApiFactory->__invoke();
+        $openApiResultArray = $normalizerInterface->normalize($openApiResult, 'json');
+        $sortedOpenApiArray = ArrayDeepSort::sort($openApiResultArray);
+        // Arguments for Yaml::dump taken from https://github.com/api-platform/core/blob/49c81194a3e6833f10d135c739776636775b15a5/src/OpenApi/Command/OpenApiCommand.php#L58
+        $openApiYaml = Yaml::dump(
+            input: $sortedOpenApiArray,
+            inline: 10,
+            indent: 2,
+            flags: Yaml::DUMP_OBJECT_AS_MAP
+            | Yaml::DUMP_EMPTY_ARRAY_AS_SEQUENCE
+            | Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK
+        );
```

Comment:
Couldn't you delegate the serialization completely to the snapshot driver? Then every future YAML snapshot we do will use the same set of arguments.

Reply:
If you delegate the YAML serialization to the driver, you could even move the ArrayDeepSort call there too.

Reply:
Its the question if all future usages want to use these arguments and want to keep the sorting.
maybe we want to test in the future, that our code preserves the order.
I can live with both, the next usage(s) will tell which way would have been smarter.